pkg/controllers/management/multiclusterapp/handler_multiclusterapp.go

```diff
@@ -43,9 +49,16 @@ func Register(ctx context.Context, management *config.ManagementContext) {
 	r := MCAppRevisionController{
 		managementContext: management,
 	}
+	projects := management.Management.Projects("")
+	p := ProjectController{
+		mcAppsLister:  management.Management.MultiClusterApps("").Controller().Lister(),
+		mcApps:        management.Management.MultiClusterApps(""),
+		projectLister: projects.Controller().Lister(),
+	}
 	m.multiClusterApps.AddHandler(ctx, "management-multiclusterapp-controller", m.sync)
 	management.Management.MultiClusterAppRevisions("").AddHandler(ctx, "management-multiclusterapp-revisions-rbac", r.sync)
 	m.prtbs.AddHandler(ctx, "management-prtb-controller-global-resource", m.prtbSync)
+	projects.AddHandler(ctx, "management-mcapp-project-controller", p.sync)
 }
```

Comment:
Don't we need a handler for clusters too? to remove target projects from that cluster?